sale_blanket_order/wizard/create_sale_orders.py

```diff
@@ -0,0 +1,205 @@
+        currency_id = 0
+        pricelist_id = 0
+        user_id = 0
+        payment_term_id = 0
+        for line in self.line_ids:
+            if line.qty == 0.0:
+                continue
+
+            if line.qty > line.remaining_uom_qty:
+                raise UserError(
+                    _('You can\'t order more than the remaining quantities'))
+
+            vals = {'product_id': line.product_id.id,
+                    'name': line.product_id.name,
+                    'product_uom': line.product_uom.id,
+                    'sequence': line.blanket_line_id.sequence,
+                    'price_unit': line.blanket_line_id.price_unit,
+                    'blanket_order_line': line.blanket_line_id.id,
+                    'product_uom_qty': line.qty,
+                    'tax_id': [(6, 0, line.taxes_id.ids)]}
+            order_lines_by_customer[line.partner_id.id].append((0, 0, vals))
+
+            if currency_id == 0:
+                currency_id = line.blanket_line_id.order_id.currency_id.id
+            elif currency_id != line.blanket_line_id.order_id.currency_id.id:
+                currency_id = False
+
+            if pricelist_id == 0:
+                pricelist_id = line.blanket_line_id.pricelist_id.id
+            elif pricelist_id != line.blanket_line_id.pricelist_id.id:
+                pricelist_id = False
+
+            if user_id == 0:
+                user_id = line.blanket_line_id.user_id.id
+            elif user_id != line.blanket_line_id.user_id.id:
+                user_id = False
+
+            if payment_term_id == 0:
+                payment_term_id = line.blanket_line_id.payment_term_id.id
+            elif payment_term_id != line.blanket_line_id.payment_term_id.id:
+                payment_term_id = False
+
+        if not order_lines_by_customer:
+            raise UserError(_('An order can\'t be empty'))
+
+        if not currency_id:
+            raise UserError(_('Can not create Sale Order from Blanket '
+                              'Order lines with different currencies'))
+
+        res = []
+        for customer in order_lines_by_customer:
+            order_vals = {
+                'partner_id': customer,
+            }
+            if self.blanket_order_id:
+                order_vals.update({
+                    'origin': self.blanket_order_id.name,
+                })
+            order_vals.update({
+                'user_id': user_id if user_id else False,
+                'currency_id': currency_id if currency_id else False,
+                'pricelist_id': pricelist_id if pricelist_id else False,
+                'payment_term_id': (payment_term_id
+                                    if payment_term_id
+                                    else False),
+                'order_line': order_lines_by_customer[customer],
+            })
+            sale_order = self.env['sale.order'].create(order_vals)
+            res.append(sale_order.id)
```

Comment:
Does this function `create_sale_order` create one sale order or can multiple sale orders be created?